src/main.qml

```diff
@@ -17,27 +17,205 @@
-        onReadingChanged: app.bpm = reading.bpm;
-        onStatusChanged: console.log("status changed to: " + str(status))
+        onReadingChanged: {
+            app.lastBpm = app.bpm
+            app.bpm = reading.bpm
+            if (app.lastBpm > 0) {
+                DisplayBlanking.preventBlanking = false
+            }
+        }
+    }
+
+    Timer {
+        id: pulseTimer
+        interval: 1000 / (app.bpm / 30)
+        running: app.bpm
+        repeat: true
+        //triggeredOnStart: true
+        onTriggered: pulseToggle ?
+                         pulseToggle = false :
+                         pulseToggle = true
+    }
+
+    Behavior on arcBpmGap { NumberAnimation { duration: 100; easing.type: Easing.Linear} }
+    Behavior on arcBpmOffset { NumberAnimation { duration: 1000; easing.type: Easing.OutCirc } }
+    Behavior on arcEnd { NumberAnimation { duration: arcAnimationDuration; easing.type: app.lastBpm === 0 ? Easing.Linear : Easing.OutInSine } }
+    Behavior on arcStart { NumberAnimation { duration: arcAnimationDuration; easing.type: app.lastBpm === 0 ? Easing.Linear : Easing.OutInSine } }
+    Behavior on pulseWidth { NumberAnimation { duration: pulseAnimationDuration; easing.type: Easing.OutInSine } }
+
+    Canvas {
+        id: canvas
+        anchors.fill: parent
+        rotation: -90
+        opacity: 0.4
+        onPaint: {
+            var ctx = getContext("2d")
+            var x = app.width / 2
+            var y = app.height / 2
+            var start = Math.PI * (arcStart / 180)
+            var end = Math.PI * (arcEnd / 180)
+            var gap1 = Math.PI * ((arcStartOffset + arcBpmOffset + 45 - arcBpmGap/2) / 180)
+            var gap2 = Math.PI * ((arcStartOffset + arcBpmOffset + 45 + arcBpmGap/2) / 180)
+
+            if (gap1 > end) gap1 = end
+            if (gap2 > end) gap2 = end
+
+            ctx.reset()
+            ctx.beginPath()
+            ctx.lineCap = "round"
+            ctx.lineWidth = app.height * 0.034
+            ctx.strokeStyle = "#38FF12"
+            ctx.arc(x, y, (app.width / 2.9), start, gap1, false)
+            ctx.stroke()
+            ctx.beginPath()
+            ctx.arc(x, y, (app.width / 2.9), gap2, end, false)
+            ctx.stroke()
+        }
     }
 
     Label {
         id: bpmText
         anchors.centerIn: parent
         horizontalAlignment: Text.AlignHCenter
-       //% "Measuring..."
-        text: app.bpm > 0 ? app.bpm : qsTrId("id-measuring")
+        font.letterSpacing: app.bpm > 0 ?
+                                -app.width * 0.008 :
+                                0
+        font.pixelSize: app.bpm > 0 ?
+                            app.bpm / 100 >= 1 ?
+                                app.height*0.29 :
+                                app.height*0.33 :
+                            app.height*0.06
+        text: app.bpm > 0 ?
+                  app.bpm :
+                  qsTrId("id-measuring")
+        Timer {
+            id: timer
```

Comment:
It would probably make sense to rename this `timer` to something more descriptive like `blinkingTimer`.